src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetEventsTest.java

```diff
@@ -0,0 +1,111 @@
+
+@AutoConfigureMockMvc
+class CaseControllerGetEventsTest  extends IntegrationBase {
+
+    public static final String EVENT_NAME = "New Description";
+    public static final String EVENT_TEXT1 = "event1";
+    @Autowired
+    private transient MockMvc mockMvc;
+
+    private static String endpointUrl = "/cases/{caseId}/hearings/{hearingId}/events";
+
+    private static final OffsetDateTime SOME_DATE_TIME = OffsetDateTime.parse("2023-01-01T12:00Z");
+    private static final String SOME_COURTHOUSE = "some-courthouse";
+    private static final String SOME_COURTROOM = "some-courtroom";
+
+    private CourtCaseEntity case1;
+    private HearingEntity hearing1;
+    private EventEntity event1;
+    private EventHandlerEntity activeHandler;
+
+    @BeforeEach
+    void setupData() {
+        CourthouseEntity swanseaCourthouse = someMinimalCourthouse();
+        swanseaCourthouse.setCourthouseName(SOME_COURTHOUSE);
+
+        CourtroomEntity courtroom1 = createCourtRoomWithNameAtCourthouse(swanseaCourthouse, SOME_COURTROOM);
+
+        case1 = createCaseAt(swanseaCourthouse);
+        case1.setCaseNumber("Case1");
+
+        JudgeEntity judge = createJudgeWithName("aJudge");
+        hearing1 = createHearingWithDefaults(case1, courtroom1, LocalDate.of(2023, 5, 20), judge);
+
+        dartsDatabase.saveAll(hearing1);
+
+        activeHandler = EventHandlerTestData.someMinimalEventHandler();
+
+        dartsDatabase.saveAll(activeHandler);
+
+        event1 = createEventWith("eventName", EVENT_TEXT1, hearing1, OffsetDateTime.now(), activeHandler);
+        EventEntity event2 = createEventWith("eventName", "event2", hearing1, OffsetDateTime.now(), activeHandler);
+        dartsDatabase.saveAll(event1, event2);
+    }
+
+    @AfterEach
+    void tearDown() {
+        dartsDatabase.addToTrash(activeHandler);
+    }
+
+    @Test
+    void casesGetEventsGetEndpoint() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = get(endpointUrl, case1.getId(), hearing1.getId());
+        mockMvc.perform(requestBuilder).andExpect(MockMvcResultMatchers.status().isOk());
+    }
+
+    @Test
+    void casesGetEventsWithInvalidCaseIdReturnsNotFoundEndpoint() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = get(endpointUrl, 789, hearing1.getId());
+        mockMvc.perform(requestBuilder).andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    @Test
+    void casesGetEventsWithInvalidHearingIdReturnsNotFoundEndpoint() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = get(endpointUrl, case1.getId(), 456);
+        mockMvc.perform(requestBuilder).andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    @Test
+    void casesGetEvents() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = get(endpointUrl, case1.getId(), hearing1.getId());
+
+        mockMvc.perform(requestBuilder).andExpect(MockMvcResultMatchers.status().isOk())
```

Comment:
Consider adding an assertion on the length of the returned array (I think expectation is length == 1).